### DISCUSSION TITLE
Including height and width of the sketch in the exported iframe tag.

### DISCUSSION THREAD

**Discussion by JithinKS97** [processing/p5.js-web-editor#2658] [General]
Existing feature enhancement

**Comment by catarak** [comment]
thanks for submitting this jithin! my opinion is that i don't think it makes sense to add this feature, but i'd like to open it up for discussion. some questions i have:

1. what if your sketch contains no canvases?
2. what if your sketch contains multiple canvases?
3. what if a user dynamically changes the width or height of their canvas?
4. what if they don't want their iframe to be the same width/height as the canvas?
5. how would we get this to work? we'd have to write a script that analyzes the sketch code (which could be in any of the files in the project), looks for `createCanvas` via a string regex (which could appear never or multiple times), and extract the numbers from there. this feels prone to breaking to me.

maybe the ui/ux could be clearer to users as to how to change the size of the iframe, and they could still enter it in manually? that feels like a better solution to me. very, very open to other ideas.

**Comment by shiffman** [comment]
I like the idea of starting with a manual option . . . we could at some point test out techniques for "suggesting" an iframe size (`createCanvas()` is one idea but could we also maybe use something like `document.body.width/height`?) but I agree there are probably too many variables.  Having the interface show clear input boxes for the user to enter width/height would be a great start.
